src/app/digital/rendering/ioobjects/WireRenderer.ts

```diff
@@ -45,7 +45,7 @@ export const WireRenderer = (() => {
             ));
                 
             // @TODO move to function for getting color based on being selection/on/off
-            const color = (wire.getIsOn() ? DEFAULT_ON_COLOR : (selected ? selectedColor : wire.getColor()));
+            const color = (wire.getInput().getIsOn() ? DEFAULT_ON_COLOR : (selected ? selectedColor : wire.getColor()));
```

Comment:
What happens if you try click on an LED's Input Port to connect the wire?
I have a feeling this might break? If not then lmk and we should be good to merge!

Reply:
You're right, it did, I didn't even think of that.
I made a quick change that I just tested and I believe will fix the issue, so let me know if it looks okay.